### ISSUE TITLE
"jQuery is not defined" JS error if jQuery moved to footer

### ISSUE THREAD

**Issue by sumobi** [AffiliateWP/AffiliateWP#2031]
If jQuery has been moved to the page footer before AffiliateWP has been installed for the very first time, meaning the `affwp_js_works` option row has not yet been created, it will cause a `ReferenceError: jQuery is not defined` JS error.

This is because the `header_scripts` method (https://github.com/AffiliateWP/AffiliateWP/blob/master/includes/class-tracking.php#L59) is loaded on `wp_head` which contains the JS for firing off the `affwp_check_js` action: https://github.com/AffiliateWP/AffiliateWP/blob/master/includes/class-tracking.php#L83-L92

Could we somehow detect where jQuery is loaded (head or footer) and make sure this is always loaded _after_?

Meanwhile here's the quickfix:

```
function affwp_custom_move_header_scripts() {
	$tracking = affiliate_wp()->tracking;

	remove_action( 'wp_head', array( $tracking, 'header_scripts' ) );
	add_action( 'wp_footer', array( $tracking, 'header_scripts' ), 100 );
}
add_action( 'template_redirect', 'affwp_custom_move_header_scripts' );
```

Ticket: https://secure.helpscout.net/conversation/319811659/53088/?folderId=205531

**Comment by DrewAPicture** [comment]
My first instinct would be to check if jQuery has been enqueued with something like `wp_script_is( 'jquery', 'enqueued' )` then set the hook based on that.

**Comment by pippinsplugins** [comment]
>>>My first instinct would be to check if jQuery has been enqueued with something like wp_script_is( 'jquery', 'enqueued' ) then set the hook based on that.

Gut reaction says that would work.

**Comment by DrewAPicture** [comment]
Added a PR: #2217

**Comment by pippinsplugins** [comment]
I haven't been able to replicate the original problem but the PR does work properly for me.

**Comment by DrewAPicture** [comment]
Merged #2217.

**Comment by DrewAPicture** [comment]
Backported to `master` via 831961560188094af1c333ce7df200c50e6cbf97. Milestone adjusted to 2.0.10.
